src/content/1.7/modules/sample_modules/order_view_page_new_hooks/signature-widget.md

```diff
@@ -0,0 +1,479 @@
+    class: PrestaShop\Module\DemoViewOrderHooks\Presenter\OrderSignaturePresenter
+    arguments:
+      - '@=service("prestashop.module.demovieworderhooks").getPathUri() ~ parameter("signatureImgDirectory")'
+
+```
+
+Let's create a twig templates in `modules/demovieworderhooks/views/templates/admin/`:
+
+card.html.twig
+
+```twig
+{% trans_default_domain 'Module.Demovieworderhooks.Admin' %}
+
+<div class="card">
+  <div class="card-header">
+    <h3 class="card-header-title">
+      {% block card_title %}
+      {% endblock %}
+    </h3>
+  </div>
+
+  <div class="card-body">
+    {% block card_body %}
+    {% endblock %}
+  </div>
+</div>
+```
+
+`customer_signature.html.twig` extending `card.html.twig`
+
+```twig
+{% extends '@Modules/demovieworderhooks/views/templates/admin/card.html.twig' %}
+
+{% block card_title %}
+  {{ 'Signature'|trans }}
+{% endblock %}
+
+{% block card_body %}
+  <div class="col-lg">
+    <div class="display-4 text-center">
+      {{ signature.gender }} {{ signature.firstName }} {{ signature.lastName }}
+    </div>
+    <div class="text-center">
+      <img src="{{ signature.imagePath }}" alt="">
+    </div>
+  </div>
+{% endblock %}
+```
+
+Lets add several methods to `DemoViewOrderHooks` class.
+`getModuleTemplatePath` - get's the path of the templates folder.
+
+```php
+    /**
+     * Get path to this module's template directory
+     */
+    private function getModuleTemplatePath(): string
+    {
+        return sprintf('@Modules/%s/views/templates/admin/', $this->name);
+    }
+```
+
+Render a twig template method:
+
+```php
+    /**
+     * Render a twig template.
+     */
+    private function render(string $template, array $params = []): string
+    {
+        /** @var Twig_Environment $twig */
+        $twig = $this->get('twig');
+
+        return $twig->render($template, $params);
+    }
+```
+
+For each registered hook, you must create a non-static public method, 
+starting with the “hook” keyword followed by the name of the hook you want to use 
+(starting with either “display” or “action”). In our case: `hookDisplayBackOfficeOrderActions`
+For more information see: https://devdocs.prestashop.com/1.7/modules/concepts/hooks/#execution
+We add this code at the bottom of the main module class `demovieworderhooks.php`
+and also add the missing `use` statements for new classes.
+
+```php
+    /**
+     * Displays customer's signature.
+     */
+    public function hookDisplayBackOfficeOrderActions(array $params)
+    {
+        /** @var OrderSignatureRepository $signatureRepository */
+        $signatureRepository = $this->get('prestashop.module.demovieworderhooks.repository.order_signature_repository');
+
+        /** @var OrderSignaturePresenter $signaturePresenter */
+        $signaturePresenter = $this->get('prestashop.module.demovieworderhooks.presenter.order_signature_presenter');
+
+        $signature = $signatureRepository->findOneBy(['orderId' => $params['id_order']]);
```

Comment:
Oh I know! You could add a `findOneByOrderId(int $orderId)` method to explain the extendability of repositories
Even though I believe doctrine already implements it via magic calls, it gives us a nice example

Reply:
yes, just both use cases are valuable; showing EntityRepository function is also same value; Sth. more practical would be better

Reply:
But if you integrate `$this->findOneBy(['orderId' => $orderId]);` you will show both features

Reply:
oh, I got, let's do that! :)

Reply:
done